### PR TITLE
CSP-558 Implements Get Apprentice Account endpoint

### PR DESCRIPTION
**Pull request by DevenKShah**

### CODE REVIEW

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application.UnitTests/ApprenticeAccount/Queries/GetApprenticeAccount/GetApprenticeAccountQueryHandlerTests.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        GetApprenticeAccountQuery request,
+        GetApprenticeAccountQueryResult expectedResult,
+        CancellationToken cancellationToken)
+    {
+
+        apiClientMock.Setup(c => c.GetWithResponseCode<GetApprenticeAccountQueryResult>(It.Is<GetApprenticeRequest>(r => r.Id == request.ApprenticeId))).ReturnsAsync(new ApiResponse<GetApprenticeAccountQueryResult>(expectedResult, HttpStatusCode.OK, null));
+
+        var actualResult = await sut.Handle(request, cancellationToken);
+
+        Assert.That(expectedResult, Is.EqualTo(actualResult));
```
Comment (ravihejmadi) [inline src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application.UnitTests/ApprenticeAccount/Queries/GetApprenticeAccount/GetApprenticeAccountQueryHandlerTests.cs:30]: Cancellelation Token verification like [this](url)?

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application.UnitTests/ApprenticeAccount/Queries/GetApprenticeAccount/GetApprenticeAccountQueryHandlerTests.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        [Frozen] Mock<IApprenticeAccountsApiClient<ApprenticeAccountsApiConfiguration>> apiClientMock,
+        GetApprenticeAccountQueryHandler sut,
+        GetApprenticeAccountQuery request,
+        CancellationToken cancellationToken)
+    {
+        apiClientMock.Setup(c => c.GetWithResponseCode<GetApprenticeAccountQueryResult?>(It.Is<GetApprenticeRequest>(r => r.Id == request.ApprenticeId))).ReturnsAsync(new ApiResponse<GetApprenticeAccountQueryResult?>(null, HttpStatusCode.NotFound, null));
+
+        var actualResult = await sut.Handle(request, cancellationToken);
+
+        Assert.That(actualResult, Is.Null);
```
Comment (ravihejmadi) [inline src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application.UnitTests/ApprenticeAccount/Queries/GetApprenticeAccount/GetApprenticeAccountQueryHandlerTests.cs:45]: No StatusCode with result? If so [Assert.AreEqual](https://github.com/SkillsFundingAgency/das-aan-hub-api/blob/b5edde3a8e2081e9738a6cad615e9869da408d03/src/SFA.DAS.AANHub.Api.UnitTests/Controllers/AdminsControllerTests.cs#L148)

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application.UnitTests/ApprenticeAccount/Queries/GetApprenticeAccount/GetApprenticeAccountQueryHandlerTests.cs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        GetApprenticeAccountQueryHandler sut,
+        GetApprenticeAccountQuery request,
+        CancellationToken cancellationToken)
+    {
+        apiClientMock.Setup(c => c.GetWithResponseCode<GetApprenticeAccountQueryResult?>(It.Is<GetApprenticeRequest>(r => r.Id == request.ApprenticeId))).ReturnsAsync(new ApiResponse<GetApprenticeAccountQueryResult?>(null, HttpStatusCode.InternalServerError, null));
+
+        Func<Task> act = () => sut.Handle(request, cancellationToken);
+
+        await act.Should().ThrowAsync<InvalidOperationException>();
+    }
```
Comment (ravihejmadi) [inline src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application.UnitTests/ApprenticeAccount/Queries/GetApprenticeAccount/GetApprenticeAccountQueryHandlerTests.cs:61]: Once again...No StatusCode with result? If so [Assert.AreEqual](https://github.com/SkillsFundingAgency/das-aan-hub-api/blob/b5edde3a8e2081e9738a6cad615e9869da408d03/src/SFA.DAS.AANHub.Api.UnitTests/Controllers/AdminsControllerTests.cs#L148)

**Review by MarkFCain — DISMISSED** [review]

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application.UnitTests/ApprenticeAccount/Queries/GetApprenticeAccount/GetApprenticeAccountQueryHandlerTests.cs (hunk 2 above)
Comment (MarkFCain) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application.UnitTests/ApprenticeAccount/Queries/GetApprenticeAccount/GetApprenticeAccountQueryHandlerTests.cs:45]: There's also an Assert.IsNull(..) I think

**Review by ravihejmadi — COMMENTED** [review]
Hi Deven, I had a quick look but it looks like we are missing GetApprenticeAccountQueryResultTests.cs Unit Tests here. I will have another look after the DS

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application.UnitTests/ApprenticeAccount/Queries/GetApprenticeAccount/GetApprenticeAccountQueryHandlerTests.cs (hunk 1 above)
Comment (DevenKShah) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Application.UnitTests/ApprenticeAccount/Queries/GetApprenticeAccount/GetApprenticeAccountQueryHandlerTests.cs:30]: Handler gets cancellation token in the argument, so don't see a need to verify that.

File: src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api/Controllers/ApprenticesController.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+    public ApprenticesController(IMediator mediator)
+    {
+        _mediator = mediator;
+    }
+
+    [HttpGet]
+    [Route("account/{apprenticeId}")]
+    [ProducesResponseType(typeof(GetApprenticeAccountQueryResult), StatusCodes.Status200OK)]
+    public async Task<IActionResult> GetAccount(Guid apprenticeId, CancellationToken cancellationToken)
```
[the remark this answers is not in the document]
Comment (DevenKShah) [reply src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api/Controllers/ApprenticesController.cs:21]: Fixed

**Review by MarkFCain — APPROVED** [review]

**Pull request merged**